simde/x86/sse4.2.h

```diff
@@ -37,6 +37,232 @@ SIMDE__BEGIN_DECLS
+    b_ = simde__m128i_to_private(b);
+  const int upper_bound = (128 / 16) - 1;
+  int a_invalid = 0;
+  int b_invalid = 0;
+  for(int i = 0 ; i < (upper_bound) ; i++) {
+    for(int j = 0; j< (upper_bound) ; j++)
+    {
+      int bitvalue = ((a_.i16[i] == b_.i16[j]) ? 1 : 0);
+      bool_res_.i16[i] |= ((bitvalue) << j);
+      if(i == la)
+        a_invalid = 1;
+      if(j == lb)
+        b_invalid = 1;
+      switch(cmp_op){
+        case SIMDE_SIDD_CMP_EQUAL_ANY:
+          break;
+        case SIMDE_SIDD_CMP_RANGES:
+          break;
+        case SIMDE_SIDD_CMP_EQUAL_EACH:
+          if(a_invalid && b_invalid)
+            bool_res_.i16[i] |= (1 << j);
+          break;
+        case SIMDE_SIDD_CMP_EQUAL_ORDERED:
+          if(a_invalid && !b_invalid)
+            bool_res_.i16[i] |= (1 << j);
+          else if(a_invalid && b_invalid)
+            bool_res_.i16[i] |= (1 << j);
+          break;
+      }
+    }
+  }
+  int32_t int_res_1 = 0;
+  int32_t int_res_2 = 0;
+  switch(cmp_op) {
+    case SIMDE_SIDD_CMP_EQUAL_ANY:
+      for(int i = 0 ; i < (upper_bound) ; i++){
+        SIMDE__VECTORIZE_REDUCTION(|:int_res_1)
+        for (int j = 0 ; j < (upper_bound) ; j++){
+          int_res_1 |= (((bool_res_.i16[i] >> j) & 1) << i) ;
+        }
+      }
+      break;
+    case SIMDE_SIDD_CMP_RANGES:
+      for(int i = 0 ; i < (upper_bound) ; i++){
+        SIMDE__VECTORIZE_REDUCTION(|:int_res_1)
+        for(int j = 0 ; j < (upper_bound) ; j++){
+          int_res_1 |= ((((bool_res_.i16[i] >> j) & 1) & ((bool_res_.i16[i] >> (j + 1)) & 1)) << i);
+          j += 2;
+        }
+      }
+      break;
+    case SIMDE_SIDD_CMP_EQUAL_EACH:
+      for(int i = 0 ; i < (upper_bound) ; i++){
+        SIMDE__VECTORIZE_REDUCTION(|:int_res_1)
+        for(int j = 0 ; j < (upper_bound) ; j++){
+          int_res_1 |= (((bool_res_.i16[i] >> i) & 1) << i);
+        }
+      }
+      break;
+    case SIMDE_SIDD_CMP_EQUAL_ORDERED:
+      int_res_1 = (imm8 & 1) ? 0xff : 0xffff;
+      for(int i = 0 ; i < (upper_bound) ; i++){
+        int k = i;
+        SIMDE__VECTORIZE_REDUCTION(|:int_res_1)
+        for(int j = 0 ; j < (upper_bound-i) ; j++){
+          int_res_1 &= (((bool_res_.i16[k] >> j) & 1) << i) ;
+          k += 1;
+        }
+      }
+  }
+  for(int i = 0; i < (upper_bound) ; i++){
+    if(polarity & 1){
+      if((polarity >> 1) & 1) {
+        if (i >= lb) {
+          int_res_2 |= (((int_res_1 >> i) & 1) << i);
+        }
+        else {
+          int_res_2 |= ((((int_res_1 >> i) & 1) ^ (-1)) << i);
+        }
+      }
+      else{
+        int_res_2 |= ((((int_res_1 >> i) & 1) ^ (-1)) << i);
+      }
+    }
+    else{
+      int_res_2 |= (((int_res_1 >> i) & 1) << i);
+    }
+  }
+  return ((int_res_2 == 0) & (lb > upper_bound));
+}
+
+SIMDE__FUNCTION_ATTRIBUTES
+int
+simde_mm_cmpestra(simde__m128i a, int la, simde__m128i b, int lb, const int imm8){
+#if defined(SIMDE_X86_SSE4_2_NATIVE)
+  return _mm_cmpestra(a, la, b, lb, imm8);
```

Comment:
Because of
``` cc
simde/x86/sse4.2.h: In function ‘simde_mm_cmpestra’:
simde/x86/sse4.2.h:253:10: error: the fifth argument must be an 8-bit immediate
   return _mm_cmpestra(a, la, b, lb, imm8);
          ^
```
You'll need to move the `SIMDE_X86_SSE4_2_NATIVE` version out to its own `define` and not inside the function definition, like in https://github.com/nemequ/simde/blob/453d5e45b0ffc4984f45bde6b43abf18df9c5ee5/simde/x86/sse4.1.h#L109

Reply:
I have corrected this too.
Thank You